test/error_messages.coffee

```diff
@@ -4,12 +4,15 @@
 # Ensure that errors of different kinds (lexer, parser and compiler) are shown
 # in a consistent way.
 
-assertErrorFormat = (code, expectedErrorFormat) ->
-  throws (-> CoffeeScript.run code), (err) ->
+assertErrorFormat = (code, expectedErrorFormat, {ast} = {}) ->
```

Comment:
It seems like a nice goal would be to have as many errors that occur when compiling to JS also get thrown when generating an AST as possible. The abstraction of `Op::checkUpdateAssignability()` that gets called from both `Op::compileNode()` and `Op::ast()` is an example of what would probably be a typical pattern for doing that

So then for testing that those errors occur when generating an AST, I tried to come up with a lightweight way to specify that existing tests should be run in the context of AST generation (in addition to compilation to JS) and expect the same errors to be thrown